policy-test/tests/outbound_api.rs

```diff
@@ -0,0 +1,391 @@
+use std::time::Duration;
+
+use futures::prelude::*;
+use kube::ResourceExt;
+use linkerd_policy_controller_k8s_api as k8s;
+use linkerd_policy_test::{create, create_service, grpc, mk_service, with_temp_ns};
+use tokio::time;
+
+#[tokio::test(flavor = "current_thread")]
+async fn service_does_not_exist() {
+    with_temp_ns(|client, ns| async move {
+        // Build a service but don't apply it to the cluster.
+        let mut svc = mk_service(&ns, "my-svc", 4191);
+        // Give it a bogus cluster ip.
+        svc.spec.as_mut().unwrap().cluster_ip = Some("1.1.1.1".to_string());
+
+        let mut policy_api = grpc::OutboundPolicyClient::port_forwarded(&client).await;
+        let rsp = policy_api.watch(&ns, &svc, 4191).await;
+
+        assert!(rsp.is_err());
+        assert_eq!(rsp.err().unwrap().code(), tonic::Code::NotFound);
+    })
+    .await;
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn service_with_no_http_routes() {
+    with_temp_ns(|client, ns| async move {
+        // Create a service
+        let svc = create_service(&client, &ns, "my-svc", 4191).await;
+
+        let mut rx = retry_watch_outbound_policy(&client, &ns, &svc).await;
+        let config = rx
+            .next()
+            .await
+            .expect("watch must not fail")
+            .expect("watch must return an initial config");
+        tracing::trace!(?config);
+
+        // There should be no routes.
+        let (h1_routes, h2_routes) = detect_http_routes(&config);
```

Comment:
I don't think this is correct. 

The Gateway API spec requires that when an HTTPRoute defines no route rules, the gateway (proxy) must return a 404. Therefore, when there are _no_ HTTPRoute resources targeting a Service, we will need to generate a default route with a rule that matches everything and has no backends/filters, so that the proxy handles traffic normally.

When there is an HTTPRoute targeting a Service, and the HTTPRoute defines no rules, we should send the proxy a route which has no rules, so that it knows to 404. It's probably worth also adding a test for that case.